driver-core/src/main/java/com/datastax/driver/core/AbstractAddressableByIndexData.java

```diff
@@ -43,75 +48,111 @@ protected ByteBuffer getValue(int i) {
     }
 
     public T setBool(int i, boolean v) {
-        checkType(i, DataType.Name.BOOLEAN);
-        return setValue(i, TypeCodec.BooleanCodec.instance.serializeNoBoxing(v));
+        checkType(i, BOOLEAN);
+        TypeCodec<Boolean> codec = codecFor(i, Boolean.class);
+        ByteBuffer bb;
+        if(codec instanceof TypeCodec.PrimitiveBooleanCodec)
+            bb = ((TypeCodec.PrimitiveBooleanCodec)codec).serializeNoBoxing(v, protocolVersion);
```

Comment:
Since the primitive codecs serializeNoBoxing methods just call serialize, is there any reason to make an exception here?  Same question for deserialize/deserializeNoBoxing.